### PR TITLE
Fix JS parsing perf issue (#7310)

### PR DESCRIPTION
**Pull request by billti**
Simple change. Description of issue/cause is in #7310.

Tests are failing on my box for unrelated issues, so I haven't run the full suite yet. Will verify on my other machine later tonight (if any failures on the CI).

### CODE REVIEW

**Comment by billti** [comment]
Ping @mhegazy , as we were looking at this earlier.

**Comment by billti** [comment]
Tests should be fixed too with the commit just pushed ( @RyanCavanaugh is right, those JSDocParser tests are a pain :disappointed: ).

File: tests/cases/unittests/jsDocParsing.ts [hunk 1]
```diff
@@ -998,7 +1001,10 @@ module ts {
                         ? JSON.parse(Utils.sourceFileToJSON(v))
                         : v;
                 }, 4);
-                
+
+                // Remove the parserContextFlags from the comparison
+                result = result.replace(/\,\n\s+\"parserContextFlags\": \"JavaScriptFile\"/g, "");
```
Comment (DanielRosenwasser) [inline tests/cases/unittests/jsDocParsing.ts]: Why is this necessary?

Comment (billti) [inline tests/cases/unittests/jsDocParsing.ts]: Because the harness cares about what flags the parser sets on the AST (see https://github.com/Microsoft/TypeScript/blob/master/src/harness/harness.ts#L277 ), but I don't care to go edit the 100s of stringified AST nodes in this file now we correctly consider this flag "*ParserGenerated*" - especially as 1). It's not meaningful for what these tests are validating, and 2). There's been talk of potentially changing the contextFlags, and I'd hate to have to edit them all again shortly down the road.

Comment (RyanCavanaugh) [inline tests/cases/unittests/jsDocParsing.ts]: Can we change `Utils.sourceFileToJSON` in `harness` to exclude that bit (~ line 279)?

**Comment by mhegazy** [comment]
LGTM. thanks!

File: tests/cases/unittests/jsDocParsing.ts (hunk 1 above)
Comment (billti) [inline tests/cases/unittests/jsDocParsing.ts]: It's not that branch, it's line 287, and the comment explicitly states *"We only care what the parser set directly on the ast"*, which these are, so it seems even more hacky (if that's possible) to explicitly drop this bit there.

**Comment by RyanCavanaugh** [comment]
:+1:
